plugins/flytekit-bigquery/flytekitplugins/bigquery/backend_plugin.py

```diff
@@ -0,0 +1,86 @@
+from typing import Dict, Optional
+
+import grpc
+from flyteidl.service.external_plugin_service_pb2 import (
+    SUCCEEDED,
+    TaskCreateResponse,
+    TaskDeleteResponse,
+    TaskGetResponse,
+)
+from google.cloud import bigquery
+
+from flytekit import FlyteContextManager, StructuredDataset
+from flytekit.core.type_engine import TypeEngine
+from flytekit.extend.backend.base_plugin import BackendPluginBase, BackendPluginRegistry, convert_to_flyte_state
+from flytekit.models import literals
+from flytekit.models.literals import LiteralMap
+from flytekit.models.task import TaskTemplate
+from flytekit.models.types import LiteralType, StructuredDatasetType
+
+pythonTypeToBigQueryType: Dict[type, str] = {
+    str: "STRING",
+    int: "INT64",
+}
+
+
+class BigQueryPlugin(BackendPluginBase):
+    def __init__(self):
+        super().__init__(task_type="bigquery_query_job_task")
+
+    def create(
+        self,
+        context: grpc.ServicerContext,
+        output_prefix: str,
+        task_template: TaskTemplate,
+        inputs: Optional[LiteralMap] = None,
+    ) -> TaskCreateResponse:
+        job_config = None
+        if inputs:
+            ctx = FlyteContextManager.current_context()
+            python_interface_inputs = {
+                name: TypeEngine.guess_python_type(lt.type) for name, lt in task_template.interface.inputs.items()
+            }
+            native_inputs = TypeEngine.literal_map_to_kwargs(ctx, inputs, python_interface_inputs)
+
+            job_config = bigquery.QueryJobConfig(
+                query_parameters=[
+                    bigquery.ScalarQueryParameter(name, pythonTypeToBigQueryType[python_interface_inputs[name]], val)
```

Comment:
worth validating the input types against https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types ?

Reply:
Thanks. updated it, but not added all the bq types here since not every type can be mapped to python type. btw, bq plugin in propeller only support 4 types. https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/plugins/webapi/bigquery/query_job.go#L218-L255